content/en/tracing/manual_instrumentation/go.md

```diff
@@ -1,27 +1,129 @@
-    - link: 'tracing/opentracing'
-      tag: 'Documentation'
-      text: 'Implement Opentracing across your applications'
     - link: 'tracing/visualization/'
       tag: 'Documentation'
       text: 'Explore your services, resources, and traces'
 ---
+This page details common use cases for adding and customizing observability with Datadog APM.
+
+## Adding Tags
+
+Add custom [span tags][1] to your [spans][2] to customize your observability within Datadog. The span tags are applied to your incoming traces, allowing you to correlate observed behavior with code-level information such as merchant tier, checkout amount, or user ID.
+
+### Add custom span tags
+
+Add [tags][1] directly to a `Span` interface by calling `SetTag`:
+
+```go
+package main
+
+import (
+    "log"
+    "net/http"
+
+    "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func handler(w http.ResponseWriter, r *http.Request) {
+    // Create a span for a web request at the /posts URL.
+    span := tracer.StartSpan("web.request", tracer.ResourceName("/posts"))
+    defer span.Finish()
+
+    // Set tag
+    span.SetTag("http.url", r.URL.Path)
+    span.SetTag("<TAG_KEY>", "<TAG_VALUE>")
+}
+
+func main() {
+    tracer.Start(tracer.WithServiceName("<SERVICE_NAME>"))
+    defer tracer.Stop()
+    http.HandleFunc("/posts", handler)
+    log.Fatal(http.ListenAndServe(":8080", nil))
+}
+```
+
+Datadog's integrations make use of the `Context` type to propagate the current active [span][2].
+If you want to add span tags attached to a `Context`, call the `SpanFromContext` function:
+
+```go
+package main
+
+import (
+    "net/http"
+
+    "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func handler(w http.ResponseWriter, r *http.Request) {
+    // Retrieve a span for a web request attached to a Go Context.
+    if span, ok := tracer.SpanFromContext(r.Context()); ok {
+        // Set tag
+        span.SetTag("http.url", r.URL.Path)
+    }
+}
+```
+
+### Adding tags globally to all spans
+
+Add [tags][1] to all [spans][2] by configuring the tracer with the `tags` option:
+
+```go
+package main
+
+import "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+func main() {
+    tracer.Start(
+        tracer.WithGlobalTag("datacenter", "us-1"),
+        tracer.WithGlobalTag("env", "dev"),
+    )
+    defer tracer.Stop()
+}
+```
+
+### Set errors on a span
+
+To set an error on one of your spans, use `tracer.WithError` as below:
+
+```go
+err := someOperation()
+span.Finish(tracer.WithError(err))
+```
 
-If you aren't using supported library instrumentation (see [Library compatibility][1]), you may want to to manually instrument your code.
+## Adding spans
 
-## Single Tracing
+If you aren't using supported library instrumentation (see [Library compatibility][3]), you may want to to manually instrument your code.
 
-To make use of manual instrumentation, use the `tracer` package which is documented on Datadog's [godoc page][2]:
+### Manually creating a new span
+
+```go
```

Comment:
This example is actually for configuring opentracing with datadog, not manually creating a span.

Spans can be created manually in a few ways, but they're used throughout these examples, such as [here](https://github.com/DataDog/documentation/pull/7728/files#diff-34d43e7ec7772fa809f9e30806e86817R37)

There are two functions that do this, [StartSpan](https://godoc.org/gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer#StartSpan) and [StartSpanFromContext](https://godoc.org/gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer#StartSpanFromContext)

Here are some possible examples:
```go
//Create a span
span := tracer.StartSpan(“mainOp”, tracer.ResourceName("/user"), tracer.ChildOf(parentSpan))

//Create a span with a resource name, which is the child of parentSpan.
span := tracer.StartSpan(“mainOp”, tracer.ResourceName("/user"), tracer.ChildOf(parentSpan))

// Create a span which will be the child of the span in the Context ctx, if there is a span in the context.
// Returns the new span, and a new context containing the new span.
span, ctx := tracer.StartSpanFromContext(ctx, “mainOp”, tracer.ResourceName("/user"))
```